gems/smithy-client/lib/smithy-client/stubbing/data_applicator.rb

```diff
@@ -18,30 +18,30 @@ def apply(data, stub)
 
         private
 
+        def shape(ref, value)
+          case ref.shape
+          when StructureShape then structure(ref.shape, value, ref.shape.type.new)
```

Comment:
Since we use structure's type here - should we do the same for unions?

Reply:
There's no union stubbing here. What did you have in mind?

Reply:
I was thinking that if you stub runtime shape for `StructureShape` -  I figured we do the same for when we want to stub `UnionShape` subclasses but I guess that is highly unlikely for that to happen. I was just curious.l